src/main/java/com/fasterxml/jackson/annotation/JsonProperty.java

```diff
@@ -57,11 +57,18 @@ public enum MyEnum {
     /**
      * Defines name of the logical property, i.e. JSON object field
      * name to use for the property. If value is empty String (which is the
-     * default), will try to use name of the field that is annotated.
+     * default), will try to use name of the field's first variable that is
+     * annotated.
      * Note that there is
      * <b>no default name available for constructor arguments</b>,
      * meaning that
      * <b>Empty String is not a valid value for constructor arguments</b>.
+     * <p>
+     * Note that
```

Comment:
Ok this is the important and useful part.

Reply:
I only provided the others, because if you were using some sort of language parsing project such as [JavaParser](http://javaparser.org/) you would have to reference it like so:

```
FieldDeclaration field;
field.getVariables().get(0);

```

It may make it less confusing to know it only applies to the first entry.

Reply:
Ah. Ok I see what you mean.